text/0000-fabric-private-chaincode.md

```diff
@@ -0,0 +1,284 @@
+ 
+In addition to the peer-side validation, clients must also be able to verify FPC transactions.
+This process is similar to the validation step as described above; the FPC client SDK extension will implement this functionality and make this process thereby transparent to the users. A low level FPC client SDK API may also provide the functionality to the end user directly.
+
+## TEE platform support
+
+Currently, our FPC Runtime and the SDK focuses on Intel SGX SDK. However, components such as the FPC Registry are already designed to support attestations by other TEE platforms as they mature and gain remote attestation capabilities. Also, other components such as the Go part of the FPC chaincode library don't have an Intel SGX depency and can easily be reused. We plan to explore other TEE platforms such as AMD SEV in the future.
+ 
+## A opportunity for Endorsement Policies
+
+Using FPC enables a new way to write Endorsement Policies. In particular, it enables to create policies which take into account that a chaincode is executed in a protected environment (i.e., TEE). Depending on the workload, this can reduce the number of required endorsements and still protect the integrity of the execution result.
+
+Other intended future work includes new possible endorsement policies that can incorporate TEE policies across multiple TEE vendors. For instance, FPC developer could write endorsement policy to express the following: two SGX endorsements OR a single IBM Z endorsement; OR 2 AMD SEV endorsements OR 5 XXX TEE endorsements;
+ 
+## Deployment Process (in detail)
+ 
+This section details the turn-up process for all elements of FPC, including an explanation of the trust architecture.
+ 
+* Step 1: Channel and Peers
+ 
+    Initially it is assumed that a standard Fabric network is in place. A new Channel is created according to the existing standard process for Fabric. One or more Peers join the channel according to the standard process. Next, each endorsing peer initiates a Ledger Enclave for the Channel, imprinting itself with a content-addressable Channel Identifier and a hash of the Channel's Genesis Block contents. Note that FPC currently does not support cross-channel interaction; however, this can be addressed in future work.
+ 
+* Step 2: Create the FPC Chaincode deployment artifact
+ 
+    One or more of the participants compiles an agreed-upon chaincode using the FPC compiler configuration. This links in the Chaincode Library, including its shim in the compiled package and creating a chaincode identifier called the MRENCLAVE. Details of this process will vary in future versions supporting other TEEs. Participants must be careful to use the same compiler configuration and version numbers of all included software. As long as they do, the MRENCLAVE will be the same for each compiled Chaincode Package.
```

Comment:
We are assuming that the chaincode is exactly the same for all organizations.

Reply:
Yes! With FPC we define a chaincode identity using MRENCLAVE.